arbitrum-client/src/conversion.rs

```diff
@@ -0,0 +1,61 @@
+//! Utilities for converting from relayer type definitions to their analogues in
+//! ther smart contracts.
+
+use constants::{SystemCommitment, SystemProof};
+use renegade_contracts_common::types::{
+    G1Affine, MatchPayload as ContractMatchPayload, Proof as ContractProof,
+};
+
+use crate::{errors::ConversionError, types::MatchPayload};
+
+/// Try to extract a fixed-length array of G1Affine points
+/// from a slice of proof system commitments
+fn try_unwrap_commitments<const N: usize>(
+    comms: &[SystemCommitment],
+) -> Result<[G1Affine; N], ConversionError> {
+    comms
+        .iter()
+        .map(|c| c.0)
+        .collect::<Vec<_>>()
+        .try_into()
+        .map_err(|_| ConversionError::InvalidLength)
+}
+
+/// Trait for converting from relayer types to their smart contract analogues
+pub trait ToContractType<T> {
+    /// Convert the type to its smart contract analogue
+    fn to_contract_type(self) -> Result<T, ConversionError>;
+}
```

Comment:
Why does this need a trait? Just use `Into`/`From`

Reply:
`Into`/`From` is a bit unwieldy here because we can't implement it directly between `mpc-jellyfish` and `renegade-contracts` types, so I would need to introduce wrapper types, which muddies the interface / readability more than I think is worth it to just use canonical traits (since we only do this conversion in one place, explicitly, anyway)

But if I'm mocking out the `renegade-contracts` types here anyway then that should no longer be an issue